google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/RateLimitingServerStreamingCallable.java

```diff
@@ -88,44 +89,164 @@ public void call(
       ((BigtableTracer) context.getTracer())
           .batchRequestThrottled(stopwatch.elapsed(TimeUnit.NANOSECONDS));
     }
-    RateLimitingResponseObserver innerObserver =
-        new RateLimitingResponseObserver(limiter, lastQpsChangeTime, responseObserver);
+    RateLimitingResponseObserver innerObserver = new RateLimitingResponseObserver(responseObserver);
     innerCallable.call(request, innerObserver, context);
   }
 
+  /** A rate limiter wrapper class that can be disabled. */
+  static class ConditionalRateLimiter {
+
+    private final AtomicBoolean enabled = new AtomicBoolean(false);
+
+    private final RateLimiter limiter;
+
+    // This is the next time allowed to change QPS or disable rate limiting.
+    private final AtomicReference<Instant> nextRateUpdateTime =
+        new AtomicReference<>(Instant.now());
+
+    public ConditionalRateLimiter(long defaultQps) {
+      limiter = RateLimiter.create(defaultQps);
+    }
+
+    /**
+     * Works the same way with {@link RateLimiter#acquire()} except that when the rate limiter is
+     * disabled, {@link ConditionalRateLimiter#acquire()} always returns 0.0 immediately.
+     */
+    public double acquire() {
+      if (enabled.get()) {
+        return limiter.acquire();
+      } else {
+        return 0;
+      }
+    }
+
+    // Enable rate limiting immediately or disable after the QPS update period. Otherwise, no-op.
+
+    /**
+     * Disables the rate limier if the current time exceeded the next rate update time. Or, enables
+     * the rate limiter immediately, setting next rate update time to now. When disabled, the rate
+     * is retained and will be re-used if re-enabled later.
+     *
+     * @param enabled Enable or disable the rate limiter.
+     */
+    public void trySetEnabled(boolean enabled) {
+      if (enabled) {
+        // Always enable immediately.
+        boolean wasEnabled = this.enabled.getAndSet(true);
+        if (!wasEnabled) {
+          logger.fine("Rate limiter is enabled.");
+        }
+        return;
+      }
+      // Only disable after the QPS update period.
+      Instant nextTime = nextRateUpdateTime.get();
+      Instant now = Instant.now();
+      if (now.isAfter(nextTime)) {
+        boolean wasEnabled = this.enabled.getAndSet(false);
+        if (wasEnabled) {
+          logger.fine("Rate limiter is disabled.");
+        }
+      }
+    }
```

Comment:
This really seems like 2 methods merged into one. I think it might be nicer to split this up into enable() and disable()

Reply:
Done.